### ISSUE TITLE
CMake doesn't find python 3.X

### ISSUE THREAD

**Issue by swertz** [MoMEMta/MoMEMta#167]
On systems with both python2 and python3 installed (which is most of them), the current cmake config is not able to find python 3. Even forcing the minimum version to 3.0 doesn't work. 

Besides, because of [PEP394](https://www.python.org/dev/peps/pep-0394/), direct calls to `python`such as https://github.com/MoMEMta/MoMEMta/blob/master/tests/bindings/python/integration_tests.py#L1 will default to python2, which is not what we want.

Specifying the paths to python libs/interpreter/includes explicitly works, but the correct version of boost-python is not found, for the same reason (libboost_python.so links to the python2 version, not to (libboost_python3.so - at least on my 'buntu).

**Comment by delaere** [comment]
googling around and experimenting, it really looks like a bug in cmake find_package(PythonLibs). Looks like they are preparing a separate find_package(Python3Libs) : https://gitlab.kitware.com/cmake/cmake/merge_requests/1819
we could maybe include these developments in our tree

**Comment by swertz** [comment]
Thanks for the link! I've tried to naively download the updated modules from the PR, but there are too many changes for that to work. Even when that is merged, there's no chance we can support only the most recent cmake version...

Actually the problem is not so much python - setting the minimum version to 3 does work, contrary to what I write above - but finding the corresponding boost python.

I've managed to get something working using the changes proposed here (but not merged): https://gitlab.kitware.com/cmake/cmake/merge_requests/35/
Putting [the modified version of FindBoost.cmake](https://gitlab.kitware.com/cmake/cmake/raw/85a0a93ce18014a0f9f7314c97d7c888a717eee7/Modules/FindBoost.cmake) into our `cmake/modules` directory and running cmake with `-DBoost_PYTHON_VERSION_TAG=-py36` worked like a charm!

So, we could ship this file until a better solution becomes mainstream, and require the user to:
- Force the min. version of python to 3 (that could easily be an additional build option)
- Specify what is the suffix for the boost::python library, since that varies from system to system. On ubuntu finding that is as easy as running `locate boost_python`, but that might be less true on other systems...

Then there's the issue of the `python` call in the test, but since the script is already being configured by cmake (see [here](https://github.com/MoMEMta/MoMEMta/blob/master/cmake/scripts/run_tests.sh.in)), that can easily be solved.
